src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-config-modal/rgw-config-modal.component.ts

```diff
@@ -152,36 +188,72 @@ export class RgwConfigModalComponent implements OnInit {
 
   onSubmit() {
     const values = this.configForm.getRawValue();
-    this.rgwBucketService
-      .setEncryptionConfig(
-        values['encryptionType'],
-        values['kms_provider'],
-        values['auth_method'],
-        values['secret_engine'],
-        values['secret_path'],
-        values['namespace'],
-        values['address'],
-        values['token'],
-        values['owner'],
-        values['ssl_cert'],
-        values['client_cert'],
-        values['client_key']
-      )
-      .subscribe({
-        next: () => {
-          this.notificationService.show(
-            NotificationType.success,
-            $localize`Updated RGW Encryption Configuration values`
-          );
-        },
-        error: (error: any) => {
-          this.notificationService.show(NotificationType.error, error);
-          this.configForm.setErrors({ cdSubmitButton: true });
-        },
-        complete: () => {
-          this.activeModal.close();
-          this.table?.refreshBtn();
+    let encryptionData: VaultConfig | KmipConfig;
+    if (values['kms_provider'] === KMS_PROVIDER.VAULT) {
+      encryptionData = {
+        kms_provider: values['kms_provider'],
+        encryption_type: values['encryptionType'],
+        config: {
+          addr: values['addr'],
+          auth: values['auth'],
+          prefix: values['secret_path'],
+          secret_engine: values['secret_engine'],
+          namespace: values['namespace'],
+          token_file: values['token'],
+          ssl_cacert: values['ssl_cert'],
+          ssl_clientcert: values['client_cert'],
+          ssl_clientkey: values['client_key']
+        }
+      };
+    } else if (values['kms_provider'] === KMS_PROVIDER.KMIP) {
+      encryptionData = {
+        kms_provider: values['kms_provider'],
+        encryption_type: values['encryptionType'],
+        config: {
+          addr: values['addr'],
+          username: values['username'],
+          password: values['password'],
+          kms_key_template: values['kms_key_template'],
+          s3_key_template: values['s3_key_template'],
+          client_key: values['client_key'],
+          ca_path: values['ssl_cert'],
+          client_cert: values['client_cert']
         }
-      });
+      };
+    }
+    this.rgwBucketService.setEncryptionConfig(encryptionData).subscribe({
+      next: () => {
+        this.notificationService.show(
+          NotificationType.success,
+          $localize`Updated RGW Encryption Configuration values`
+        );
+      },
+      error: (error: Error) => {
+        this.notificationService.show(NotificationType.error, error.message);
+        this.configForm.setErrors({ cdSubmitButton: true });
+      },
+      complete: () => {
+        this.activeModal.close();
+        this.table?.refreshBtn();
+      }
+    });
+  }
+
+  setKmsProvider() {
```

Comment:
we could get the kmsProvider as the input here and use that for comparison
```suggestion
  setKmsProvider(kmsProvider: string) {
```

Reply:
and btw, what is the job of this function. looking by what i see in the function, its definitely not just for setting kms provider because that is happening above somewhere as I pointed out earlier. Let's name the function properly so that its easier to deduce from the name itself the responsibility of a function.

Reply:
changed the function functionality